lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -202,7 +202,10 @@ class ObjectLockInfo {
      * @returns {bool} - True if the retention policy allows the objects data to be modified (overwritten/deleted)
      */
     canModifyObject(hasGovernanceBypass) {
-        return !this.isLocked() || (this.isGovernanceMode() && !!hasGovernanceBypass);
+        // can modify object if object is not locked
+        // cannot modify object in any cases if legal hold is enabled
+        // if not legal hold, can only modify object if bypass governance when locked
+        return !this.legalHold && (!this.isLocked() || (this.isGovernanceMode() && !!hasGovernanceBypass));
```

Comment:
isLocked function already checks the legalHold
. was wondering why keep it in the second part of the condition ?

Reply:
yeah isLockes contains legalhold check but if we remove the second part then `!this.legalHold && (this.isGovernanceMode() && !!hasGovernanceBypass)`  will allow any compliance or governance mode without legalhold pass which should not be allowed